client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -626,13 +644,12 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response) {
                 }
             }
         }
-        LoggerHelpers.traceLeave(log, "getSuccessorsFromCut", traceId);
-        return CompletableFuture.completedFuture(unread);
+        return unread;
     }
-    
-    private List<Segment> computeKnownUnreadSegments(StreamSegments currentSegments, StreamCut from) {
+
+    private List<Segment> computeKnownUnreadSegments(Collection<Segment> currentSegments, StreamCut from) {
```

Comment:
since the logic has changed from current segments to a `toStreamCut` this computation would not be correct.